src/rgw/rgw_crypt.cc

```diff
@@ -403,12 +404,13 @@ class AES_256_CBC : public BlockCrypt {
   static const size_t AES_256_KEYSIZE = 256 / 8;
   static const size_t AES_256_IVSIZE = 128 / 8;
   static const size_t CHUNK_SIZE = 4096;
+  const DoutPrefixProvider* dpp;
```

Comment:
@alimaredia instead of declaring DPP here can't we make the `AES_256_CBC` class a DoutPrefixProvider?

Reply:
somehow that seems a little wierd

Reply:
@cbodley do you think the BlockCrypt class is fine to have a DPP member or should we make it a provider?

Reply:
lets avoid multiple inheritance if possible.
IMO, "composition" should be the default, and "inheritance" (without virtual functions) needs strong justification

Reply:
Yeah, it's not always wrong, but in this case, we'd be heading for a design in which many objects are DPPs, and that's sailing in the direction of "everything is an Object" territory